api/accessrequest/access_request.go

```diff
@@ -0,0 +1,177 @@
+/*
+Copyright 2023 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package accessrequest
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/client"
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/api/types"
+)
+
+type ListResourcesRequestOption func(*proto.ListResourcesRequest)
+
+// GetResourcesWithFilters is a wrapper for client.GetResourcesWithFilters
+// that maps resource kinds to the resource types expected by ListResources.
+func GetResourcesWithFilters(ctx context.Context, clt client.ListResourcesClient, req proto.ListResourcesRequest, kind string) ([]types.ResourceWithLabels, error) {
```

Comment:
I think we need a better name for this. It's not obvious to me by reading the name or the godocs when this should be preferred over calling `clt.GetReosourcesWithFilters` directly.